machine/v1/types_nutanixprovider.go

```diff
@@ -154,6 +166,161 @@ type NutanixResourceIdentifier struct {
 	Name *string `json:"name,omitempty"`
 }
 
+// NutanixGPUIdentifierType is an enumeration of different resource identifier types for GPU entities.
+type NutanixGPUIdentifierType string
+
+const (
+	// NutanixGPUIdentifierName identifies a GPU by Name.
+	NutanixGPUIdentifierName NutanixGPUIdentifierType = "Name"
+
+	// NutanixGPUIdentifierDeviceID identifies a GPU by device ID.
+	NutanixGPUIdentifierDeviceID NutanixGPUIdentifierType = "DeviceID"
+)
+
+// NutanixGPU holds the identity of a Nutanix GPU resource in the Prism Central
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'DeviceID' ?  has(self.deviceID) : !has(self.deviceID)",message="deviceID configuration is required when type is DeviceID, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Name' ?  has(self.name) : !has(self.name)",message="name configuration is required when type is Name, and forbidden otherwise"
+// +union
+type NutanixGPU struct {
+	// type is the identifier type to use for this GPU resource.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NutanixGPUIdentifierType `json:"type"`
+
+	// deviceID is the id of the GPU entity.
+	// +optional
+	// +unionMember
+	DeviceID *int64 `json:"deviceID,omitempty"`
+
+	// name is the GPU name
+	// +optional
+	// +unionMember
+	Name *string `json:"name,omitempty"`
+}
+
+// NutanixStorageContainerReferenceType is an enumeration of different storage_container reference types.
+// +kubebuilder:validation:Enum:=UUID;Name
+type NutanixStorageContainerReferenceType string
+
+const (
+	// NutanixStorageContainerReferenceUUID is a reference type identifying the storage_container object by UUID.
+	NutanixStorageContainerReferenceUUID NutanixStorageContainerReferenceType = "UUID"
+
+	// NutanixStorageContainerReferenceName is a reference type identifying the storage_container object by Name.
+	NutanixStorageContainerReferenceName NutanixStorageContainerReferenceType = "Name"
+
+	// NutanixStorageContainerReferenceURL is a reference type identifying the storage_container object by URL.
+	NutanixStorageContainerReferenceURL NutanixStorageContainerReferenceType = "URL"
```

Comment:
This one isn't in the enum?

Reply:
Removed "URL". We will not support it now.